scripts/Muon/GUI/Common/load_run_widget/model.py

```diff
@@ -0,0 +1,97 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import (absolute_import, division, print_function)
+
+from mantid.api import WorkspaceGroup
+
+from Muon.GUI.Common.muon_load_data import MuonLoadData
+import Muon.GUI.Common.utilities.load_utils as load_utils
+
+
+def is_workspace_group(workspace):
+    return isinstance(workspace, WorkspaceGroup)
+
+
+def get_run_from_multi_period_data(workspace_list):
+    """
+    Checks if multi-period data has a single consistent
+    run number and returns it, otherwise raises ValueError.
+    """
+    runs = [ws.getRunNumber() for ws in workspace_list]
+    unique_runs = list(set(runs))
+    if len(unique_runs) != 1:
+        raise ValueError("Multi-period data contains >1 unique run number.")
+    else:
+        return unique_runs[0]
+
+
+def exception_message_for_failed_files(failed_file_list):
+    return "Could not load the following files : \n - " + "\n - ".join(failed_file_list)
+
+
+class LoadRunWidgetModel(object):
+    """Stores info on all currently loaded workspaces"""
+
+    def __init__(self, loaded_data_store=MuonLoadData()):
+        # Used with load thread
+        self._filenames = []
+
+        self._loaded_data_store = loaded_data_store
+        self._current_run = None
+
+    # TODO : remove the need for this function
```

Comment:
do we want this to do?

Reply:
I can't quite work out why it is isn't needed currently.